database/transaction.go

```diff
@@ -170,6 +203,16 @@ func (tx Transaction) CreateIndex(opts IndexConfig) error {
 	return tx.indexStore.Insert(opts)
 }
 
+// UpdateIndex updates the index cfg referenced by cfg.IndexName.
+func (tx Transaction) UpdateIndex(cfg IndexConfig) error {
```

Comment:
I don't think this should be exported, it seems really specific to the renaming feature and can be dangerous in other contexts. There doesn't seem to be a reason to replace an index config other than for renaming its related table.

Reply:
Yeah very good point!